### PR TITLE
[HAL-1832] Activate OIDC in the console

### PR DESCRIPTION
**Pull request by hpehl**
https://issues.redhat.com/browse/HAL-1832

### CODE REVIEW

File: console/HAL-1832_Activate_OIDC.adoc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+Affected UI:: Configuration / Subsystems / Elytron OIDC Client
+Affected Resources:: `/subsystem=elytron-oidc-client`
+
+=== Hard Requirements
+
+The console shows the new `elytron-oidc-client` subsystem. The UI for this subsystem is backed by the https://hal.github.io/documentation/concepts/#applications[model browser] which generates a UI based on the resource descriptions.
+
+The UI makes it possible to configure the necessary resources as described in the Elytron admin guide: `docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc`, section "Securing the management console with OpenID Connect".
+
+If the console is secured by the Keycloak OpenID Connect (OIDC) adapter, the "Access Control" top level category shows a screen which summarizes the basic settings. This screen, contains also a link to the Keycloak admin console.
```
Comment (OndrejKotek) [inline console/HAL-1832_Activate_OIDC.adoc:71]: > If the console is secured by the Keycloak OpenID Connect (OIDC) adapter, ...

I believe you mean _secured by Keycloak OpenID Connect (OIDC) provider_, as the client part is covered by the native support.

File: console/HAL-1832_Activate_OIDC.adoc [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+The console does not offer a dedicated UI/wizards to configure the `elytron-oidc-client` subsystem. This might be addressed in a future RFE.
+
+== Open Questions
+
+* None
+
+== Test Plan
+
+Additional tests are added to the test suite that verify that the hash is visible in the deployment preview.
```
Comment (OndrejKotek) [inline console/HAL-1832_Activate_OIDC.adoc:87]: Are you going to add the new tests to the new Berg test suite?

Comment (hpehl) [reply console/HAL-1832_Activate_OIDC.adoc:87]: Well, I thought, you would be able to add the tests 😬

File: console/HAL-1832_Activate_OIDC.adoc (hunk 1 above)
Comment (hpehl) [reply console/HAL-1832_Activate_OIDC.adoc]: Thanks, I fixed that and added a bit more context.

File: console/HAL-1832_Activate_OIDC.adoc (hunk 2 above)
Comment (OndrejKotek) [reply console/HAL-1832_Activate_OIDC.adoc:91]: OK :)

File: console/HAL-1832_Activate_OIDC.adoc (hunk 1 above)
Comment (OndrejKotek) [reply console/HAL-1832_Activate_OIDC.adoc]: Looks better now, thanks a lot.

File: console/HAL-1832_Activate_OIDC.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
```
Comment (OndrejKotek) [inline console/HAL-1832_Activate_OIDC.adoc:91]: > Additional tests are added to the test suite that verify that the hash is visible in the deployment preview.

I haven't read this thoroughly, now I can't imagine what meant by this/what will be verified (which hash, which deployment preview)

File: console/HAL-1832_Activate_OIDC.adoc [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+== Requirements
+
+Affected UI:: Configuration / Subsystems / Elytron OIDC Client
+Affected Resources:: `/subsystem=elytron-oidc-client`
+
+=== Hard Requirements
+
+The console shows the new `elytron-oidc-client` subsystem. The UI for this subsystem is backed by the https://hal.github.io/documentation/concepts/#applications[model browser] which generates a UI based on the resource descriptions.
+
+The UI makes it possible to configure the necessary resources as described in the Elytron admin guide: `docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Elytron_OIDC_Client.adoc`, section "Securing the management console with OpenID Connect". In particular, this means the console makes it possible to add the following resources:
```
Comment (kstekovi) [inline console/HAL-1832_Activate_OIDC.adoc:69]: Is possible configure RBAC via HAL console?
```
/core-service=management/access=authorization:write-attribute(name=provider,value=rbac)
/core-service=management/access=authorization:write-attribute(name=use-identity-roles,value=true)
```

Comment (OndrejKotek) [reply console/HAL-1832_Activate_OIDC.adoc:69]: This is actually a good question as the scenario with RBAC `access-control` is valid in this context, see the documented CLI steps from the base RFE: https://github.com/wildfly/wildfly/pull/16856/files#diff-82d49f155de7505904b2e963e7ecb31f2ab37132f5fb93c2e74dddb6d5a7497dR347

File: console/HAL-1832_Activate_OIDC.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+* `/subsystem=elytron-oidc-client/provider=keycloak`
+* `/subsystem=elytron-oidc-client/secure-deployment=wildfly-management`
+* `/subsystem=elytron-oidc-client/secure-server=wildfly-console`
+
+If the console is secured by the Keycloak OpenID Connect (OIDC) provider, the "Access Control" top level category shows a summary of the basic settings. This contains a link to the Keycloak admin console.
```
Comment (OndrejKotek) [inline console/HAL-1832_Activate_OIDC.adoc:75]: The "Access Control" top level category is not visible when RBAC `access-control` is set.

Comment (darranl) [reply console/HAL-1832_Activate_OIDC.adoc:75]: Does the authenticated user have permission to configure access control themselves once roles are mapped?

Comment (kstekovi) [reply console/HAL-1832_Activate_OIDC.adoc:75]: yes, I had user with `ADMINISTRATOR` role.  I didn't saw the "Access Control" tab when RBAC was correctly configured. 

There is 7 predefined roles which user can have assigned (monitor, operator, maintainer, deployer, superuser, adminsitrator, auditor)

File: console/HAL-1832_Activate_OIDC.adoc (hunk 4 above)
Comment (hpehl) [reply console/HAL-1832_Activate_OIDC.adoc:69]: Yes, it is possible to configure RBAC in the console.

Comment (kstekovi) [reply console/HAL-1832_Activate_OIDC.adoc:69]: Where i can find this setup? Is there also possibility to configure `use-identity-roles`?

Comment (hpehl) [reply console/HAL-1832_Activate_OIDC.adoc:69]: CleanShot 2023-06-05 at 10 49 38

To modify the attribute `use-identity-roles` you need to use the generic model browser:
CleanShot 2023-06-05 at 11 01 54

**Pull request merged**